extension/ts/workers/finder/impl/LibGen.ts

```diff
@@ -0,0 +1,53 @@
+import {Finder, FinderParameters, FinderResponse} from "../finder";
+import {getLinks} from "./util/getLinks";
+
+const MAX = 3;
+
+const BASE_URL = "https://libgen.is";
+
+const getUrl = (base: string, queryKey: string, maxLenQuery: number, parameters: FinderParameters) =>
+	`${base}${ new URLSearchParams({[queryKey]: shorten(maxLenQuery, parameters)})}`;
+
+const simplify = (text: string): string =>
+	text
+		.replace(/[-'`~!@#$%^&*()_|+=?;:",.<>{}[\]\\/]/gi, ' ')
+		.replace(/\s+/g, ' ');
+
+const shorten = (maxLenQuery: number, {author, title}: FinderParameters): string => {
+	const shorterAuthor = simplify(author);
+	const shorterTitle = simplify(title);
+	const combined = `${shorterAuthor} ${shorterTitle}`;
+	if (combined.length > maxLenQuery) {
+		const maxLengthText = combined.slice(0, maxLenQuery).trim();
+		return maxLengthText.replace(/\s[^\s]*$/, "");
+	} else {
+		return combined;
+	}
+};
```

Comment:
were you running into issues with this while testing?

Reply:
hmm I will document this.

LibGen has a limit on its non fiction searches of 80 characters.